lib/vagrant/environment.rb

```diff
@@ -564,6 +565,11 @@ def push(name)
           pushes: push_registry.keys
       end
 
+      # Validate the global push configuration and our local push configuration
+      machine = self.machine(self.machine_names.first, self.default_provider)
+      machine.action_raw(:config_validate, Vagrant::Action::Builtin::ConfigValidate)
+      config.validate(machine)
```

Comment:
With that, then our validation right now will validate it properly. I also am not sure if its proper for the `#push` method to validate the push, since the `#machine` method doesn't validate the machine. I think it makes sense back in the `push` command.